ddtrace/contrib/kafka/patch.py

```diff
@@ -104,6 +114,7 @@ def traced_produce(func, instance, args, kwargs):
         service=trace_utils.ext_service(pin, config.kafka),
         span_type=SpanTypes.WORKER,
     ) as span:
+        span.set_tags(pin.tags)
```

Comment:
If this is only one additional tag (that we know is a string value), we would be better of using `set_tag_str`.

Reply:
The `set_tag_str` is not overloaded like `set_tag` (which `set_tags` calls), so is the more optimal call if you know the value type is always a string.

I think though your use of pin tags is worth considering. I think pins should be used for user overrides of default configurations of an integration. But this change appears to be how to capture information when the patched producer is initialized so that we can add it to the producer span. Or put differently, this pin tag should not be set by a user nor would we want to override the value in the config argument when the producer is initialized.

Reply:
I was trying to address @mabdinur 's [comment](https://github.com/DataDog/dd-trace-py/pull/6036#discussion_r1232593245) - is there another mechanism I can use to set the tag at `TracedProducer.__init__` before the span gets created?

Reply:
Yes, sorry about this. We should avoid using `Span.set_tag` and `Span.set_tags` internally. Unfortunalty we already do this in a few places. 

I would prefer reverting to the previous implementation were we stored the bootstrap_servers in a field and then set it using `set_tag_str()`. Sorry about the back and forth.

Reply:
Aside from that this PR looks good to me